bfvmm/src/exit_handler/test/test_exit_handler_intel_x64.cpp

```diff
@@ -2446,6 +2449,8 @@ exit_handler_intel_x64_ut::test_vmwrite_failure()
 
     RUN_UNITTEST_WITH_MOCKS(mocks, [&]
     {
-        EXPECT_EXCEPTION(eh->dispatch(), std::runtime_error);
+        auto f = [&] { eh->dispatch(); };
+        auto e = std::shared_ptr<std::exception>(new std::runtime_error("exit handler vmwrite failed"));
```

Comment:
Why not use `std::make_shared`?

I'd love to see this simplified to something like
```
this->expect_exception([&]{ eh->dispatch() }, std::make_shared<std::runtime_error>("error));
```

What would be even cooler is if there was an overload that allowed us to pass in the function you wanted to call without having to wrap it in a lambda. 

Thoughts?